django/db/backends/base/operations.py

```diff
@@ -652,3 +655,17 @@ def window_frame_rows_start_end(self, start=None, end=None):
 
     def window_frame_range_start_end(self, start=None, end=None):
         return self.window_frame_rows_start_end(start, end)
+
+    def explain_query_prefix(self, format=None, **options):
+        if not self.connection.features.supports_explaining_query_execution:
+            raise NotSupportedError('This backend does not support explaining query execution.')
+
+        if format:
+            supported_formats = self.connection.features.supported_explain_formats
+            if format.upper() not in supported_formats:
+                msg = '{0} is not a recognised format.'.format(format.upper())
+                if supported_formats:
+                    msg += ' Allowed formats: {0}'.format(', '.join(supported_formats))
+                raise ValueError(msg)
+
```

Comment:
What about erroring if `options` is non-empty at this point? Subclasses should have already consumed their arguments from it, and if there's anything left it's probably a mistake, like `explain(formatt='json')`

Reply:
I'm a bit on the fence about this, on one hand it would be nice for this to be pretty flexible with what it accepts as I think it's primarily going to be used as a debugging helper in a REPL, but if you're passing in options to any backend except Postgres then something is going wrong.

In any case I've made it throw a `RuntimeError`, let me know what you think

Reply:
Best to be conservative and error rather than silently swallow mistyped arguments :) I think `ValueError` or `TypeError` are more appropriate, they're normally used for argument validation.